pkg/consolegraphql/accesscontroller/access_controller.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ *
+ */
+
+package accesscontroller
+
+import (
+	"github.com/enmasseproject/enmasse/pkg/consolegraphql/cache"
+	authv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/kubernetes"
+)
+
+type AccessController interface {
+	CanRead(obj interface{}) (bool, error)
+	ViewFilter() cache.ObjectFilter
+}
+
+type AccessControlledResource interface {
+	GetControllingResourceAttributes() *authv1.ResourceAttributes
+}
+
+type allowAll struct {
+}
+
+func NewAllowAllAccessController() AccessController {
+	return &allowAll{}
+}
+
+func (a *allowAll) CanRead(obj interface{}) (bool, error) {
+	return true, nil
+}
+
+func (a *allowAll) ViewFilter() cache.ObjectFilter {
+	return func(obj interface{}) (bool, bool, error) {
+		return true, true, nil
+	}
+}
+
+type kubernetesRbac struct {
+	clientSet         kubernetes.Interface
+	cachedPermissions map[authv1.ResourceAttributes]bool
```

Comment:
Needs a sync.Mutex or RWLock to guard this as you mention in the TODO.

Reply:
Yeah - that's my idea.  The AccessController instance will be per HTTP session so concurrency through this object instance will be low (user agents can make concurrent HTTP requests for the same browsing session).  I this a Mutex will suffice.

Reply:
AccessController ended up being single threaded with state injected into them.  I believe this answers the threading concerns.